core/src/rpc.rs

```diff
@@ -382,6 +382,84 @@ impl JsonRpcRequestProcessor {
         Ok(result)
     }
 
+    pub fn get_inflation_reward(
+        &self,
+        addresses: Vec<Pubkey>,
+        epoch: Option<Epoch>,
+    ) -> Result<Vec<Option<RpcInflationReward>>> {
+        let commitment = Some(CommitmentConfig::finalized());
+        let epoch_schedule = self.get_epoch_schedule();
+        let first_available_block = self.get_first_available_block();
+        let epoch = epoch.unwrap_or_else(|| {
+            epoch_schedule
+                .get_epoch(self.get_slot(commitment))
+                .saturating_sub(1)
+        });
+
+        // Rewards for this epoch are found in the first confirmed block of the next epoch
+        let first_slot_in_epoch = epoch_schedule.get_first_slot_in_epoch(epoch.saturating_add(1));
+        if first_slot_in_epoch < first_available_block {
+            if self.bigtable_ledger_storage.is_some() {
+                return Err(RpcCustomError::LongTermStorageSlotSkipped {
+                    slot: first_slot_in_epoch,
+                }
+                .into());
+            } else {
+                return Err(RpcCustomError::BlockCleanedUp {
+                    slot: first_slot_in_epoch,
+                    first_available_block,
+                }
+                .into());
+            }
+        }
+
+        let first_confirmed_block_in_epoch = *self
+            .get_confirmed_blocks_with_limit(first_slot_in_epoch, 1, commitment)?
+            .get(0)
+            .ok_or(RpcCustomError::BlockNotAvailable {
+                slot: first_slot_in_epoch,
+            })?;
+
+        let first_confirmed_block = if let Ok(Some(first_confirmed_block)) = self
+            .get_confirmed_block(
+                first_confirmed_block_in_epoch,
+                Some(RpcConfirmedBlockConfig::rewards_only().into()),
+            ) {
+            first_confirmed_block
+        } else {
+            return Err(RpcCustomError::BlockNotAvailable {
+                slot: first_confirmed_block_in_epoch,
+            }
+            .into());
+        };
+
+        let mut reward_hash: HashMap<String, Reward> = HashMap::new();
+        for reward in first_confirmed_block
+            .rewards
+            .unwrap_or_default()
+            .into_iter()
+        {
+            reward_hash.insert(reward.clone().pubkey, reward);
+        }
```

Comment:
Can we filter this iterator to just the addresses of interest so we don't have to reallocate all the rewards?